web/src/lib/components/NostrChat.svelte

```diff
@@ -154,9 +178,84 @@
         })
     }
 
+    function queryNoteInformationInBatches() {
+        let noteInfoToGetLocal: string[] = [];
+
+        let i=0;
+
+        for (const [key, note] of notesMap) {
+            if (note === null) {
+                notesMap.set(key, true);
+                noteInfoToGetLocal.push(key);
+                i++;
+
+                if (i == nostrQueriesBatchSize) {
+                    break;
+                }
+            }
+        }
+
+        if (noteInfoToGetLocal.length === 0) {
+            return;
+        }
+
+        pool.relays.forEach(relay => {
+            let filter: Filter = {
+                kinds: [
+                    nostrEventKinds.note,
+                    nostrEventKinds.replies,
+                    nostrEventKinds.reactions,
+                ],
+                '#e': noteInfoToGetLocal
+            };
+
+            const sub: Sub = relay.sub([filter]);
+
+            sub.on('event', event => {
+                const kind = event.kind;
+
+                // TODO: REPLIES AND LIKES
+
+                if (kind === 7) {
+                    const id = event.tags.reverse().find((tag: any) => tag[0] === 'e')?.[1]; // last e tag is the liked post
+                    const eventReaction: string = event.content;
+                    const eventPubkey: string = event.pubkey;
+
+                    if (!id) {
+                        console.error('EVENT WITHOUT ID !!!');
+                        return;
+                    }
+
+                    for (let message of messages) {
+                        if (message.id === id) {
+                            let reactions: Map<string, Set<string>> | undefined = message.reactions;
+                            if (reactions === undefined) {
```

Comment:
This seems rather hacky. Can't we always initialize message.reactions in the Message constructor?